File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/model/AlertSeverity.java
package com.skedgo.tripkit.common.model;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

import static com.skedgo.tripkit.common.model.RealtimeAlert.SEVERITY_ALERT;
import static com.skedgo.tripkit.common.model.RealtimeAlert.SEVERITY_WARNING;

@StringDef({SEVERITY_WARNING, SEVERITY_ALERT})
@Retention(RetentionPolicy.SOURCE)
public @interface AlertSeverity {
}

File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/model/alert/AlertSeverity.kt
package com.skedgo.tripkit.common.model.alert

import androidx.annotation.StringDef
import com.skedgo.tripkit.common.model.realtimealert.RealtimeAlert
import kotlin.annotation.AnnotationRetention.SOURCE

@StringDef(RealtimeAlert.SEVERITY_WARNING, RealtimeAlert.SEVERITY_ALERT)
@Retention(SOURCE)
annotation class AlertSeverity 